### PR TITLE
ISPN-15269 Create or Update a role

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-15269

### CODE REVIEW

**Comment by karesti** [comment]
unrelated failures

File: cli/src/main/java/org/infinispan/cli/commands/rest/Roles.java [hunk 1]
```diff
@@ -132,6 +132,33 @@ protected CompletionStage<RestResponse> exec(ContextAwareCommandInvocation invoc
       }
    }
 
+   @CommandDefinition(name = "update", description = "Updates an existing role")
```
Comment (tristantarrant) [inline cli/src/main/java/org/infinispan/cli/commands/rest/Roles.java:135]: You need to add `Roles.Update.class` to the `@GroupCommandDefinition` at the top

**Review by tristantarrant — CHANGES_REQUESTED** [review]
Please modify `user.adoc` in the CLI help to include these changes.

**Comment by tristantarrant** [comment]
```
 FATAL [o.i.SERVER] ISPN080028: Infinispan Server failed to start org.infinispan.rest.framework.RegistrationException: ISPN012015: Cannot register invocation 'ROLES UPDATE': resource already registered for method 'PUT' at the destination path '/{role}'
 ```

**Comment by karesti** [comment]
@tristantarrant green

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**